pkgs/by-name/wh/whitakers-words/package.nix

```
@@ -0,0 +1,77 @@
+{ stdenvNoCC
+, lib
+, gnat
+, gprbuild
+, fetchFromGitHub
+, writeShellScriptBin
+, symlinkJoin
+}:
+
+let
+  pkg = stdenvNoCC.mkDerivation rec {
+    pname = "whitakers-words";
+    version = "1.97FC";
+
+    src = fetchFromGitHub {
+      owner = "mk270";
+      repo = "whitakers-words";
+      rev = "9b11477e53f4adfb17d6f6aa563669dc71e0a680";
+      sha256 = "sha256-f/8dQff2min0FivBKTk/kcwwoW5IfajAjsdkALT52xU=";
+    };
+
+    nativeBuildInputs = [
+      gprbuild
+      gnat
+    ];
+
+    dontConfigure = true;
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/{bin,lib/${pname}}
+
+      cp -r bin $out/lib/${pname}/bin
+      cp -r lib $out/lib/${pname}/lib
+      cp *.GEN $out/lib/${pname}/
+      cp *.LAT $out/lib/${pname}/
+      cp *.SEC $out/lib/${pname}/
```

```suggestion
      mkdir -p $out/bin $out/lib/whitakers-words

      cp -r bin $out/lib/whitakers-words/bin
      cp -r lib $out/lib/whitakers-words/lib
      cp *.GEN $out/lib/whitakers-words/
      cp *.LAT $out/lib/whitakers-words/
      cp *.SEC $out/lib/whitakers-words/
```
substituting pname like this is an anti pattern

---

Will fix! If I may ask though, why is that? Is there somewhere I can read about it, or design patterns in general?

---

For pname specifically #277994